app/scripts/web/app.py

```
@@ -0,0 +1,2160 @@
+    if not assignment_text:
+        lib_path_str = (request.form.get("assignment_library_path") or "").strip()
+        if lib_path_str:
+            lib_p = Path(lib_path_str)
+            if lib_p.exists():
+                ext_lp = lib_p.suffix.lower()
+                if ext_lp == ".pdf":
+                    try:
+                        import fitz
+                        doc = fitz.open(str(lib_p))
+                        assignment_text = "\n".join(page.get_text() for page in doc).strip()
+                    except Exception as exc:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)